sql/maintenance_utils.sql

```diff
@@ -29,7 +29,40 @@ CREATE OR REPLACE FUNCTION decompress_chunk(
     if_compressed BOOLEAN = false
 ) RETURNS REGCLASS AS '@MODULE_PATHNAME@', 'ts_decompress_chunk' LANGUAGE C STRICT VOLATILE;
 
-CREATE OR REPLACE FUNCTION recompress_chunk(
-    chunk REGCLASS,
-    if_not_compressed BOOLEAN = false
-) RETURNS REGCLASS AS '@MODULE_PATHNAME@', 'ts_recompress_chunk' LANGUAGE C STRICT VOLATILE;
+-- Recompress a chunk
+--
+-- Will give an error if the chunk was not already compressed. In this
+-- case, the user should use compress_chunk instead.
+--
+-- Parameters:
+--   chunk: Chunk to recompress.
+--   if_not_compressed: Print notice instead of error if chunk is already compressed.
+CREATE OR REPLACE PROCEDURE recompress_chunk(chunk REGCLASS,
+                                             if_not_compressed BOOLEAN = false)
+AS $$
+DECLARE
+  status INT;
+  chunk_name TEXT[];
+BEGIN
+    status := _timescaledb_internal.chunk_status(chunk);
```

Comment:
Could we use SQL instead to get the status?
select ch.status INTO status
from _timescaledb_catalog.chunk ch, pg_class 
where oid = '_timescaledb_internal._hyper_1_1_chunk'::regclass and relname = ch.table_name;

Reply:
Does this work for distributed hypertables?

Reply:
@gayyappan It is significantly easier and less error-prone to to maintain the `chunk_status` function than it is to maintain a fairly complex SQL query. In addition, it should probably be

```sql
SELECT ch.status INTO status
  FROM _timescaledb_catalog.chunk ch, pg_class cl, pg_namespace ns
 WHERE cl.oid = chunk AND relnamespace = ns.oid
   AND nspname = ch.schema_name  AND relname = ch.table_name AND NOT ch.dropped;
```

Which sort of proves my point that it is hard to maintain and will not adapt well to changes in the schema.

Reply:
I was referring to Gayathri's comment. Just wondering if `select ch.status INTO status
from _timescaledb_catalog.chunk ch, pg_class
where oid = '_timescaledb_internal._hyper_1_1_chunk'::regclass and relname = ch.table_name;` would work for distributed hypertables.